sources/api/migration/migrator/src/main.rs

```diff
@@ -349,14 +237,38 @@ where
     let mut intermediate_datastores = HashSet::new();
 
     for migration in migrations {
+        // get the migration from the repo
+        let lz4_bytes = repository
+            .read_target(migration.as_str())
+            .context(error::LoadMigration {
+                migration: migration.to_owned(),
+            })?
+            .context(error::MigrationNotFound {
+                migration: migration.to_owned(),
```

Comment:
```suggestion
                migration,
            })?
            .context(error::MigrationNotFound {
                migration,
```
You shouldn't need to convert to owned yourself, Snafu does that, and only as necessary.  (If you're doing a clone, or something else that needs a few cycles, you should aim for `.with_context(|| bla)` instead of a plain context, so it only happens in the failure case and not always.  But this case is so common that Snafu does it.)